aws_helper/config.go

```diff
@@ -83,10 +83,33 @@ func CreateAwsSession(config *AwsSessionConfig, terragruntOptions *options.Terra
 	} else if terragruntOptions.IamRole != "" {
 		sess.Config.Credentials = stscreds.NewCredentials(sess, terragruntOptions.IamRole, credentialsOptFn)
 	}
+	return sess, nil
+}
+
+// Returns an AWS session object for the given config region (required), profile name (optional), and IAM role to assume
+// (optional), ensuring that the credentials are available. If the config is nil, then this will return the default
+// session credentials.
+func CreateAwsSession(config *AwsSessionConfig, terragruntOptions *options.TerragruntOptions) (*session.Session, error) {
+	var sess *session.Session
+	var err error
+	if config == nil {
+		sess, err = session.NewSession()
+		if err != nil {
+			return nil, errors.WithStackTrace(err)
+		}
+		if terragruntOptions.IamRole != "" {
+			sess.Config.Credentials = stscreds.NewCredentials(sess, terragruntOptions.IamRole)
+		}
```

Comment:
I see the IAM role being used here, but do we need to do something with the profile here? I think both IAM role and profile are handled automatically in `CreateAwsSessionFromConfig` in the `else` block below, but the profile seems to be missing from this `if` block?

Reply:
I must be missing something obvious. We don't have support for AWS profiles specified via the CLI or config (which is what gets translated to `terragruntOptions`), but instead rely on the SDK environment variable `AWS_PROFILE`, which is handled via `session.NewSession`. Did I miss a struct attribute?

Reply:
Ok I think there might have been confusion from the functions docs, so I took a stab at updating it. Does it make more sense what is going on now?

861668f

Reply:
Ah, yea, the docs were indicating there was a profile we needed to handle, but with the clarification, it looks like that should already have been baked into the passed in `config`.